guide/src/components/ComponentDoc.js

```diff
@@ -1,108 +1,79 @@
-import React from 'react';
-import R from 'ramda';
-import Scroll from 'react-scroll';
-import theme from '../theme';
-import {
-    Hr,
-    P,
-    Title,
-    Div,
-    Section
-} from 'cyverse-ui';
-import {
-    Header,
-    SideNav,
-    Figure,
-    ThemeExamples,
-    MDBlock
-} from './';
+import React from "react";
+import R from "ramda";
+import Scroll from "react-scroll";
+import theme from "../theme";
+import { Hr, P, Title, Div, Section } from "cyverse-ui";
+import { Header, SideNav, Figure, ThemeExamples, MDBlock } from "./";
 
 const ScrollAnchor = Scroll.Element;
 
 class ComponentDoc extends React.Component {
-    tableData = (prop) => {
+    tableData = prop => {
         const defaultValue = prop[1].defaultValue;
-        const renderDefault = defaultValue ? defaultValue.value: "";
+        const renderDefault = defaultValue ? defaultValue.value : "";
         return (
-            <tr key={ prop[0] }>
+            <tr key={prop[0]}>
+                <td>{prop[0]}</td>
                 <td>
-                    { prop[0] }
+                    <code>{prop[1].type.name}</code>
                 </td>
                 <td>
-                    <code>
-                        { prop[1].type.name }
-                    </code>
-                </td>
-                <td>
-                    <code>
-                        { renderDefault }
-                    </code>
-                </td>
-                <td>
-                    { prop[1].description }
+                    <code>{renderDefault}</code>
                 </td>
+                <td>{prop[1].description}</td>
             </tr>
         );
     };
 
     render() {
         const { meta, children } = this.props;
         const { description, displayName } = meta;
+        const anchorId = displayName
+            .replace(/(^[A-Z])/, ([first]) => first.toLowerCase())
+            .replace(
+                /([A-Z])/g,
+                ([letter]) => `-${letter.toLowerCase()}`
+            );
```

Comment:
Are we guaranteed that `this.props.meta.displayName`  will have a value? 
Could it be `null` or `undefined`?

Reply:
I think the answer is "it'll always have a value". But in the case of the older `React.createClass({...})` definitions, if I didn't provided `displayName` and I did *not* assign the result of `React.createClass({...})` to a variable, then React DevTools wouldn't know the "display name". 

My thoughts here are "legacy" ... or "in the past". But just asking (for a friend).

Reply:
This will throw an error. I let it because it would force me to include in the component. This is a lazy way to do this though and should probably throw a proper error message like "SomeNewComponent doesn't have a displayName. The displayName is required for documentation" I'll write an issue.

Reply:
Okay.